runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/ActiveWorkState.java

```diff
@@ -219,14 +220,12 @@ synchronized ImmutableList<RefreshableWork> getRefreshableWork(Instant refreshDe
         .collect(toImmutableList());
   }
 
-  private void incrementActiveWorkBudget(Work work) {
-    activeGetWorkBudget.updateAndGet(
-        getWorkBudget -> getWorkBudget.apply(1, work.getSerializedWorkItemSize()));
+  private synchronized void incrementActiveWorkBudget(Work work) {
+    activeGetWorkBudget = activeGetWorkBudget.apply(1, work.getSerializedWorkItemSize());
   }
 
-  private void decrementActiveWorkBudget(Work work) {
-    activeGetWorkBudget.updateAndGet(
-        getWorkBudget -> getWorkBudget.subtract(1, work.getSerializedWorkItemSize()));
+  private synchronized void decrementActiveWorkBudget(Work work) {
```

Comment:
looks like this can deadlock, checking.

Reply:
how about removing this file from the PR and we can merge the rest?

in invalidateStuckCommits we call a function under the sychronized block which could be contributing to deadlock. Or could be printActiveWork holding synchronized as it calls into Work

Reply:
I suspected deadlock, but it was the n^2 loop inside failWorkForKey() holding the lock for too long that caused the delay. 

This PR is not locking more mutexes than before, all calls to `decrementActiveWorkBudget`, `incrementActiveWorkBudget`, `currentActiveWorkBudget ` are done holding the ActiveWorkStae.this mutex. 

Should be good to merge this change.